src/functions/module_math.ts

```diff
@@ -1357,6 +1363,114 @@ export const SQRT = {
+    functionCode: Maybe<FunctionResultObject>,
+    ...refs: Matrix<{ value: string }>[]
+  ) {
+    let code = toInteger(functionCode, this.locale);
+    let acceptHiddenCells = true;
+    if (code > 100) {
+      code -= 100;
+      acceptHiddenCells = false;
+    }
+    if (code < 1 || code > 11) {
+      return new EvaluationError(
+        _t("The function code (%s) must be between 1 to 11 or 101 to 111.", code)
+      );
+    }
+
+    const evaluatedCellToKeep: EvaluatedCell[] = [];
+
+    for (const ref of refs) {
+      const ref0 = ref[0][0];
+      const sheetName = splitReference(ref0.value).sheetName;
+      const sheetId = sheetName ? this.getters.getSheetIdByName(sheetName) : this.__originSheetId;
+
+      if (!sheetId) continue;
+      const { top, left } = toZone(ref0.value);
+      const right = left + ref.length - 1;
+      const bottom = top + ref[0].length - 1;
+
+      for (let row = top; row <= bottom; row++) {
+        if (this.getters.isRowFiltered(sheetId, row)) continue;
+        if (!acceptHiddenCells && this.getters.isRowHiddenByUser(sheetId, row)) continue;
+
+        for (let col = left; col <= right; col++) {
+          const cell = this.getters.getCell({ sheetId, col, row });
+          if (!cell || !cell.isFormula || !cell.content || !cell.content.includes("SUBTOTAL")) {
+            evaluatedCellToKeep.push(this.getters.getEvaluatedCell({ sheetId, col, row }));
+          }
+        }
+      }
+    }
+
+    return functionRegistry
+      .get(subtotalFunctionAggregateByCode[code])
+      .compute.apply(this, [[evaluatedCellToKeep]]);
+  },
+  isExported: true,
+} satisfies AddFunctionDescription;
+
+const subtotalFunctionAggregateDescriptions = [
+  (fnName: string) => _t("%s (accept hidden cells)", fnName),
+  (fnName: string) => _t(`%s (doesn't accept hidden cells)`, fnName),
+];
+
+autoCompleteProviders.add("subtotal_function_code_aggregate", {
+  sequence: 50,
+  autoSelectFirstProposal: true,
+  getProposals(tokenAtCursor) {
+    const functionContext = tokenAtCursor.functionContext;
+    if (
+      !functionContext ||
+      functionContext.parent.toUpperCase() !== "SUBTOTAL" ||
+      functionContext.argPosition !== 0
+    ) {
+      return;
+    }
+    const subtotalFunctionCodes = Object.keys(subtotalFunctionAggregateByCode);
+    if (subtotalFunctionCodes.includes(tokenAtCursor.value)) {
+      return;
+    }
+
+    const proposals: any[] = [];
+
+    for (let i = 0; i < subtotalFunctionAggregateDescriptions.length; i++) {
+      for (const functionCode of subtotalFunctionCodes) {
+        const str = `${i === 1 ? Number(functionCode) + 100 : functionCode}`;
+        proposals.push({
+          text: str,
+          description: subtotalFunctionAggregateDescriptions[i](
+            subtotalFunctionAggregateByCode[functionCode]
+          ),
+          htmlContent: [{ value: str, color: tokenColors.NUMBER }],
+          fuzzySearchKey: str,
+          alwaysExpanded: true,
+        });
+      }
+    }
```

Comment:
Something like that would be a lot easier to read IMO, with a small helper function `codeToProposal`
```ts
    for (const code of subtotalFunctionCodes) {
      proposals.push(codeToProposal(code, _t("%s (accept hidden cells)", subtotalFunctionAggregateByCode[code])));
    }
    for (const code of subtotalFunctionCodes.map((c) => `${Number(c) + 100}`)) {
      proposals.push(codeToProposal(code, _t("%s (doesn't accept hidden cells)", subtotalFunctionAggregateByCode[code])));
    }
```